mynn/layers/batchnorm.py

```diff
@@ -0,0 +1,121 @@
+import numpy as np
+import mygrad as mg
+
+from mynn.initializers.constant import constant
+
+class batchnorm:
+    ''' A batch normalization layer.
+
+    This class will perform an n-dimensional batch normalization operation on an (N, D, ...)
+    shaped tensor scaled by γ of shape (D, ...) and shifted by β of shape (D, ...).
+    '''
+    def __init__(self, input_shape, momentum=0.1, gamma_initializer=constant, 
+                 beta_initializer=constant, moving_mean_initializer=constant, 
+                 moving_var_initializer=constant, gamma_kwargs={"value": 1.0}, 
+                 beta_kwargs=None):
+        ''' Initialize a batch normalization layer.
+
+        Parameters
+        ----------
+        input_shape : Sequence[int]
+            The shape of the data to be batch-normalized.
+
+        momentum : float, optional (default=0.1)
+            The momentum value used to maintain moving averages.
+
+        gamma_initializer : Callable, optional (default=initializers.constant)
+            The function to use to initialize the gamma tensor.
+
+        beta_initializer : Callable, optional (default=initializers.constant)
+            The function to use to initialize the beta tensor. 
+
+        moving_mean_initializer : Callable, optional (default=initializers.constant)
+            The function to use to initialize the moving mean tensor. 
+
+        moving_var_initializer : Callable, optional (default=initializers.constant)
+            The function to use to initialize the moving variance tensor. 
+
+        gamma_kwargs : Optional[dictionary] (default={"value": 1.0})
+            The keyword arguments to pass to the gamma initialization function.
+            Initially set to pass a constant value of 1.0 to the scaling factor.
+
+        beta_kwargs : Optional[dictionary]
+            The keyword arguments to pass to the beta initialization function.
+        '''
+        self.gamma = gamma_initializer(*input_shape, 
+            **(gamma_kwargs if gamma_kwargs is not None else {}))
+        self.beta = beta_initializer(*input_shape, 
```

Comment:
`self.beta = mg.Tensor(np.zeros((1, input_channels), dtype=np.float32))` for the moment